### PR TITLE
Feature/cmw 587/resolve vulnerabilities

### PR DESCRIPTION
**Pull request by koltsov-iv**
Resolves: vulnerabilities issue

### Checklist

- [ ] Code is properly formatted
- [ ] All commits are signed
- [ ] Tests included/updated or not needed
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: types/keypair/secp256k1/public_key.go [hunk 1]
```diff
@@ -3,66 +3,57 @@ package secp256k1
...
 
-// used to reject malleable signatures
-// see:
-//   - https://github.com/ethereum/go-ethereum/blob/f9401ae011ddf7f8d2d95020b7446c17f8d98dc1/crypto/signature_nocgo.go#L90-L93
-//   - https://github.com/ethereum/go-ethereum/blob/f9401ae011ddf7f8d2d95020b7446c17f8d98dc1/crypto/crypto.go#L39
-var secp256k1halfN = new(big.Int).Rsh(btcec.S256().N, 1)
-
-type PublicKey btcec.PublicKey
+type PublicKey struct {
+	key *secp256k1.PublicKey
```
Comment (Vladimir-Kuchinskiy) [inline types/keypair/secp256k1/public_key.go:15]: why using pointer here ?

**Review by Vladimir-Kuchinskiy — COMMENTED** [review]
@koltsov-iv check my comment pls

File: types/keypair/secp256k1/public_key.go (hunk 1 above)
Comment (koltsov-iv) [reply types/keypair/secp256k1/public_key.go:15]: github.com/decred/dcrd/dcrec/secp256k1/v4 library implement methods by pointer

**Review by Vladimir-Kuchinskiy — APPROVED** [review]
lgtm

**Pull request merged**